packages/process/process_monitor.pony

```diff
@@ -197,16 +199,26 @@ actor ProcessMonitor
 
   var _closed: Bool = false
 
-  new create(notifier: ProcessNotify iso, filepath: FilePath,
-    args: Array[String] val, vars: Array[String] val)
+  new create(auth: ProcessMonitorAuth, notifier: ProcessNotify iso,
+    filepath: FilePath, args: Array[String] val, vars: Array[String] val)
   =>
     """
     Create infrastructure to communicate with a forked child process
     and register the asio events. Fork child process and notify our
     user about incoming data via the notifier.
     """
     _notifier = consume notifier
-    if not filepath.caps(FileExec) then
+
+    // We need permission to execute and the
+    // file itself needs to be an executable
+    let ok =
+    try
+      filepath.caps(FileExec) and FileInfo(filepath).mode.any_exec
+    else
+      false
+    end
+
+    if not ok then
       _notifier.failed(this, CapError)
```

Comment:
should this be an error other than CapError if the FilePath points to a non executable file?